File: Easy/graffiti-on-the-fence/graffiti-on-the-fence.js
const L = parseInt(readline());
const N = parseInt(readline());
let results = [[0, L]];

for (let i = 0; i < N; i++) {
    var inputs = readline().split(' ');
    const st = parseInt(inputs[0]);
    const ed = parseInt(inputs[1]);
    let newResults = [];

    results.forEach(result => {
        if (ed <= result[0] || st >= result[1]){
            newResults.push(result);
        }
        else if (st <= result[0] && ed >= result[0] && ed < result[1]){
            newResults.push([ed, result[1]]);
        }
        else if (st > result[0] && ed < result[1]){
            newResults.push([result[0], st]);
            newResults.push([ed, result[1]]);
        }
        else if (st > result[0] && st < result[1] && ed >= result[1]){
            newResults.push([result[0], st]);
        }
    });

    results = newResults

}

results.forEach(result => console.log(`${result[0]} ${result[1]}`))

if (results.length === 0) console.log("All painted")
 

File: Easy/graffiti-on-the-fence/graffiti-on-the-fence.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

class Solution
{
    static void Main(string[] args)
    {
        int L = int.Parse(Console.ReadLine());
        int N = int.Parse(Console.ReadLine());

        List<List<int>> results = new List<List<int>>{new List<int>{0,L}};

        for (int i = 0; i < N; i++)
        {
            string[] inputs = Console.ReadLine().Split(' ');
            int st = int.Parse(inputs[0]);
            int ed = int.Parse(inputs[1]);
            List<List<int>> newResults = new List<List<int>>();
            
            results.ForEach((result) => {
                if (ed <= result[0] || st >= result[1])
                {
                    newResults.Add(result);
                }
                else if (st <= result[0] && ed >= result[0] && ed < result[1])
                {
                    newResults.Add(new List<int>{ed, result[1]});
                }
                else if (st > result[0] && ed < result[1])
                {
                    newResults.Add(new List<int>{result[0], st});
                    newResults.Add(new List<int>{ed, result[1]});
                }
                else if (st > result[0] && st < result[1] && ed >= result[1])
                {
                    newResults.Add(new List<int>{result[0], st});
                }
            });

            results = newResults;
        }

        results.ForEach((result) => Console.WriteLine($"{result[0]} {result[1]}"));

        if (results.Count == 0) Console.WriteLine("All painted");
    }
} 
